### DISCUSSION TITLE
Something strange with metrics

### DISCUSSION THREAD

**Discussion by max851** [concourse/concourse#8776] [General]
We have a standard configuration check resource interval (1 min) and about 600 resources. I expect a rate of checks to ~10 per second (almost all is a git or docker resources).
But lidar metrics (for example `concourse_lidar_checks_started`) show only ~3.5 checks per second.
I made a recheck by collecting VPC flow. It shows ~9 requests per second (accepted connection to git or docker registry).

So the question: is metric broken or it's something that should be reconfigured on my side?

**Comment by evanchaoli** [comment]
Do you turn on global resources? If yes, when multiple resources have the exact same `source`, then only 1 check will run.

**Comment by max851** [comment]
Hello @evanchaoli 
Yes. `--enable-global-resources` enabled. 
The weird thing is that the traffic statistics show more connection than the metric. So, is the metric correct and I do a wrong interpretation of the traffic?
Or vice versa.

**Comment by pn-santos** [comment]
The nr of connections will also include the connections used to pull/push from/to the git/docker registry (e.g. when a task runs, the worker might need to pull the image, when a `get` or `put` is performed the worker will also need to establish connections).

Basically are you ensuring that _only_ resource checks are running? or are jobs also running? if the latter then the additional connections would be from pulling/pushing stuff during job runs.

**Comment by max851** [comment]
Yes. You definitely right, my pipelines do this. 
So, if i use for example two registry resource in a job then operation like `get` from first -> some change -> `put` to second:
metric should show only 2 connection per minute but is it should exist 4 connection in real life(in case if the job is running)?

As i read upper then only 2 checks should ran (global resources is enabled, Concourse 7.9.1, containerd). Or it's true for resource's check only, not for `get`/`put` parts of job?

The side question: Is there a different metric which can show complete requests rate to external resources? (or sum/combination from few metrics)
